src/tools/ceph_objectstore_tool.cc

```diff
@@ -239,12 +232,9 @@ struct pgid_object_list {
         cout << std::endl;
       }
       f->open_array_section("pgid_object");
-      string pgid = i->first.c_str();
-      std::size_t pos = pgid.find("_");
-      if (pos == string::npos)
-        f->dump_string("pgid", pgid);
-      else
-        f->dump_string("pgid", pgid.substr(0, pos));
+      spg_t pgid;
+      if (i->first.is_pg(&pgid))
+        f->dump_string("pgid", stringify(pgid));
```

Comment:
I see you've improved dump(), so I'll need to think about how to make that work.  This is a little special in the tool because it needs to know how to parse the output of "list" or soon "meta-list" as the identifier of an object/file.